wagtail/signal_handlers.py

```diff
@@ -117,6 +121,9 @@ def register_signal_handlers():
     post_save.connect(reset_locales_display_names_cache, sender=Locale)
     post_delete.connect(reset_locales_display_names_cache, sender=Locale)
 
+    post_delete.connect(clear_localemanager_cache, sender=Locale)
+    post_save.connect(clear_localemanager_cache, sender=Locale)
```

Comment:
Issue: I'm afraid this cache clearing won't work quite as expected. Signals are only run in the thread that's handling the request, which means the cache isn't cleared in those threads.

The solution to this is to either use Redis as the cache, or architect the cache such that it doesn't need clearing (which may not be possible given data can change at any time).

Reply:
I think the first step here is to figure out how big a problem it is when a Locale has been deleted, but continues to be supplied, or a Locale has been updated, but the old version continues to be supplied:

- I don't think lookups by ID are too much of a problem, as `locale_id` values on objects retrieved from the database would not be referencing deleted objects
- Items cached by `language_code` are a bigger problem, but I don't think we need to worry about the 'default' Locale ever changing (it's created automatically via a migration, and changing it would cause much bigger problems). That leaves us with `get_active()` and custom `get_for_language_code()` cases... which ARE worth some more thought.

Reply:
I suppose the biggest risk is that someone attempts to save an object with a reference to a locale that no longer exists in the database, but I'm sure we can find a way to help prevent that.

Reply:
And maybe we can warn people off changing the `language_code` of an existing `Locale`, or perhaps even prevent it completely? That sounds like reasonable behaviour to me.